### PR TITLE
Feature towards sqla2

### PR DESCRIPTION
**Pull request by bodik**
project cleanup:

* add github actions ci
* drop support for old sqlachemy and python
* add sqlalchemy 1.4 support
* add inferring model names from sqla1.4 (2.0 style) select object

### CODE REVIEW

File: sqlalchemy_filters/models.py [hunk 1]
```diff
@@ -152,13 +187,21 @@ def auto_join(query, *model_names):
...
 
     for name in model_names:
         model = get_model_class_by_name(model_registry, name)
-        if model not in get_query_models(query).values():
+        if model and (model not in get_query_models(query).values()):
             try:
-                query = query.join(model)
+                tmp = query.join(model)
+                if sqlalchemy_version_cmp('>=', '1.4'):  # pragma: nocover
+                    tmp._compile_state()
```
Comment (ArkhipovNikita) [inline sqlalchemy_filters/models.py]: In SQLAlchemy >= 1.4 'Select' object has no attribute '_compile_state', so I suppose here the following should be 
```
if sqlalchemy_version_cmp('<', '1.4'):  # pragma: nocover
    tmp._compile_state()
```

**Review by ArkhipovNikita — CHANGES_REQUESTED** [review]

File: sqlalchemy_filters/models.py (hunk 1 above)
Comment (bodik) [reply sqlalchemy_filters/models.py]: the compiling state is there for a reason (https://docs.sqlalchemy.org/en/14/changelog/migration_14.html#many-core-and-orm-statement-objects-now-perform-much-of-their-construction-and-validation-in-the-compile-phase) and it works properly in other cases, but ..

... I guess that the test I've added does not trigger auto_join so I don't see it's actually still a bit broken for select object

Comment (bodik) [reply sqlalchemy_filters/models.py]: hopefully 2696d7c82c06d0e90dfaa3406db3bc9860e2a9b2 would fix the things, @ArkhipovNikita ?

Comment (ArkhipovNikita) [reply sqlalchemy_filters/models.py]: Yes, now everything is great.

**Pull request merged**